lib/ansible/module_utils/facts.py

```diff
@@ -1208,58 +1213,122 @@ def get_dmi_facts(self):
                 else:
                     self.facts[k] = 'NA'
 
-    @timeout(10)
-    def get_mount_facts(self):
-        uuids = dict()
-        self.facts['mounts'] = []
+    def _run_lsblk(self, lsblk_path):
+        args = ['--list', '--noheadings', '--paths',  '--output', 'NAME,UUID']
+        cmd = [lsblk_path] + args
+        rc, out, err = self.module.run_command(cmd)
+        return rc, out, err
+
+    def _lsblk_uuid(self):
+
+        uuids = {}
+        lsblk_path = self.module.get_bin_path("lsblk")
+        if not lsblk_path:
+            return uuids
+
+        rc, out, err = self._run_lsblk(lsblk_path)
+        if rc != 0:
+            return uuids
+
+        lsblk_lines = out.splitlines()
+
+        # each line will be in format:
+        # <devicename><some whitespace><uuid>
+        # /dev/sda1  32caaec3-ef40-4691-a3b6-438c3f9bc1c0
+        for lsblk_line in lsblk_lines:
+            if not lsblk_line:
+                continue
+
+            line = lsblk_line.strip()
+            fields = line.split()
```

Comment:
The comment on line 1235 says that we'll always have [filepath][whitespace][uuid].  Is that true?  If so we should calculate fields like this:
``` python
fields = line.rsplit(1)
```
That way we'll handle systems which might have spaces in the filepath portion.